dotnet/src/Azure.Iot.Operations.Services/AssetAndDeviceRegistry/IAdrServiceClient.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.Iot.Operations.Services.AssetAndDeviceRegistry.AdrBaseService;
+using Azure.Iot.Operations.Services.AssetAndDeviceRegistry.AepTypeService;
+
+namespace Azure.Iot.Operations.Services.AssetAndDeviceRegistry;
+
+public interface IAdrServiceClient : IAsyncDisposable
+{
+    Task<NotificationResponse> ObserveAssetEndpointProfileUpdatesAsync(string aepName,
+        Func<string, AssetEndpointProfile?, Task> onAssetEndpointProfileUpdateEventReceived,
+        CancellationToken cancellationToken);
+
+    Task<NotificationResponse> UnobserveAssetEndpointProfileUpdatesAsync(string aepName,
+        CancellationToken cancellationToken);
+
+    Task<AssetEndpointProfile?> GetAssetEndpointProfileAsync(string aepName,
+        TimeSpan? commandTimeout = null,
+        CancellationToken cancellationToken = default);
+
+    Task<AssetEndpointProfile?> UpdateAssetEndpointProfileStatusAsync(string aepName,
+        UpdateAssetEndpointProfileStatusRequestPayload requestPayload,
+        TimeSpan? commandTimeout = null,
+        CancellationToken cancellationToken = default);
+
+    Task<NotificationResponse> ObserveAssetUpdatesAsync(string aepName,
+        string assetName,
+        Func<string, Asset?, Task> onAssetUpdateEventReceived,
+        CancellationToken cancellationToken);
+
+    Task<NotificationResponse> UnobserveAssetUpdatesAsync(string aepName,
+        string assetName,
+        CancellationToken cancellationToken);
+
+    Task<Asset?> GetAssetAsync(string aepName,
+        GetAssetRequestPayload requestPayload,
+        TimeSpan? commandTimeout = null,
+        CancellationToken cancellationToken = default);
+
+    Task<Asset?> UpdateAssetStatusAsync(string aepName,
+        UpdateAssetStatusRequestPayload requestPayload,
+        TimeSpan? commandTimeout = null,
+        CancellationToken cancellationToken = default);
+
+    Task<CreateDetectedAssetResponseSchema?> CreateDetectedAssetAsync(string aepName,
+        CreateDetectedAssetRequestPayload requestPayload,
+        TimeSpan? commandTimeout = null,
+        CancellationToken cancellationToken = default);
+
+    Task<CreateDiscoveredAssetEndpointProfileResponseSchema?> CreateDiscoveredAssetEndpointProfileAsync(string aepName,
+        CreateDiscoveredAssetEndpointProfileRequestPayload requestPayload,
+        TimeSpan? commandTimeout = null,
+        CancellationToken cancellationToken = default);
```

Comment:
I think these APIs shouldn't be exposed to users. Creating assets/asset endpoint profiles is something that only the operator should be doing in this context. 

The only reason they are in the DTDL is so that the service implementation can use codegen as well.

Reply:
From the discussion with the service folks, it sounds like this can stay as-is after all